noir-projects/aztec-nr/aztec/src/macros/notes/mod.nr

```diff
@@ -377,7 +377,21 @@ comptime fn generate_multi_scalar_mul(
 ///             hiding_point
 ///         }
 ///     }
-/// }
+///
+///     fn encrypt_log(self, context: &mut PrivateContext, recipient_keys: aztec::protocol_types::public_keys::PublicKeys, recipient: aztec::protocol_types::address::AztecAddress) -> [Field; 17] {
```

Comment:
I moved the encryption functionality to the SetupPayload because it allowed me to compute the `encrypted_log_field_length` at comptime and a user did not need to deal with that.